src/base/memory_monitor_sr.cpp

```diff
@@ -358,25 +439,88 @@ namespace cubperf
     return 0;
   }
 
-  int memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size, bool expand)
+  void memory_monitor::add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size, bool is_expanded)
   {
-    return 0;
+    LOG_TDES *tdes;
+    TRANID trid = NULL_TRANID;
+    int module_idx = MMON_GET_MODULE_IDX_FROM_STAT_ID (stat_id);
+    int comp_map_idx = MMON_GET_COMP_MAP_IDX_FROM_STAT_ID (stat_id);
+    int idx_map_val, comp_idx, subcomp_idx;
+    bool allocation_at_init = false;
+
+    if (!LOG_ISRESTARTED ())
+      {
+	allocation_at_init = true;
+      }
+
+    idx_map_val = m_module[module_idx]->get_comp_idx_map_val (comp_map_idx);
+    comp_idx = get_comp_idx_from_comp_idx_map (idx_map_val);
+    subcomp_idx = get_subcomp_idx_from_comp_idx_map (idx_map_val);
+
+    m_total_mem_usage += size;
+
+    m_module[module_idx]->add_stat (size, comp_idx, subcomp_idx, allocation_at_init, is_expanded);
+
+    if (thread_p)
```

Comment:
Is it possible for `thread_p` to have a NULL value?
if not, assertion is required for this. 
`assert (thread_p != NULL);`

Reply:
or use `LOG_FIND_THREAD_TRAN_INDEX()`

Reply:
If there is some cases that memory allocation with no thread_p parameter in function.

Reply:
Even the function you mentioned can retrieve `thread_p` because it can find the `thread_p` on which it is currently executing. 
Hmm... I have a question. 
Are you planning to track memory usage only for user transactions? If memory usage tracking is not required for system transactions, I agree to allow `thread_p` to have a NULL value. And if `thread_p` is provided, we would need to filter out cases where the `thread_p` represents a system transaction.

Reply:
OK. I change LOG_FIND_TDES to LOG_FIND_CURRENT_TDES.